### PR TITLE
chore: bump deps

### PR DESCRIPTION
**Pull request by DmitriyMV**
- run rekres
- bump github.com/ProtonMail/gopenpgp/v2 to v2.7.4
- bump github.com/klauspost/compress to v1.17.2
- bump github.com/siderolabs/go-retry to v0.3.3
- bump go.etcd.io/bbolt to v1.3.8
- bump go.uber.org/goleak to v1.3.0
- bump golang.org/x/sync to v0.4.0
- bump google.golang.org/grpc to v1.59.0
- bump actions/checkout to v4.1.1
- bump docker/setup-qemu-action to v3.0.0
- bump docker/setup-buildx-action to v3.0.0
- bump docker/login-action to v3.0.0

### CODE REVIEW

File: .github/workflows/ci.yml
Comment (frezbo) [inline .github/workflows/ci.yml:1]: rekres shouldn't put these values?

Comment (DmitriyMV) [reply .github/workflows/ci.yml:1]: Its from
https://github.com/cosi-project/runtime/pull/351
https://github.com/cosi-project/runtime/pull/330
https://github.com/cosi-project/runtime/pull/329
https://github.com/cosi-project/runtime/pull/328

Comment (smira) [reply .github/workflows/ci.yml:1]: it's not managed by Kres here

Comment (smira) [reply .github/workflows/ci.yml:1]: it runs on GitHub runners

Comment (frezbo) [reply .github/workflows/ci.yml:1]: ahh, should we move to ours?

Comment (smira) [reply .github/workflows/ci.yml:1]: it was built as a community project, independent of the Sidero Labs infra

**Pull request merged**